File: eval/test-cases/inheritance/js/src/tester.js
import GetDocumentByIdRequest from './generated/api/GetDocumentByIdRequest.js';
import PostDocumentRequest from './generated/api/PostDocumentRequest.js';
import Document from './generated/model/Document.js';
import Letter from './generated/model/Letter.js';
import Certificate from './generated/model/Certificate.js';
import ApiException from './generated/exception/ApiException.js';

const TEST_OUTPUT_MARKER="#+#"

let successes = [];
let fails = [];

async function getLetterById() {
    try {
        const letter = await GetDocumentByIdRequest.getDocumentById(1);
        if (
            letter.id === 1
            && letter.destination === "Germany"
            && letter.documentType === "LETTER"
        ) {
            successes.push("getLetterById");
        } else {
            fails.push("getLetterById:Received letter did not match expected values");
        }
    } catch (e) {
        fails.push("getLetterById:" + e);
    }
}

async function getCertificateById() {
    try {
        const certificate = await GetDocumentByIdRequest.getDocumentById(2);
        if (
            certificate.id === 2
            && certificate.certificateHolder === "Maria Mustermann"
            && certificate.documentType === "CERTIFICATE"
        ) {
            successes.push("getCertificateById");
        } else {
            fails.push("getCertificateById:Received certificate did not match expected values");
        }
    } catch (e) {
        fails.push("getCertificateById:" + e);
    }
}

async function getDocumentByUnknownId() {
    try {
        await GetDocumentByIdRequest.getDocumentById(3);
        fails.push("getDocumentByUnknownId:Request should have led to error but did not.");
    } catch (e) {
        if (e instanceof ApiException) {
            successes.push("getDocumentByUnknownId");
        } else {
            fails.push("getDocumentByUnknownId:" + e);
        }
    }
}

async function saveLetter() {
    try {
        const letter = new Letter(3, "Mars", "LETTER");
        await PostDocumentRequest.postDocument(letter);
        successes.push("saveLetter");
    } catch (e) {
        fails.push("saveLetter:" + e);
    }
}

async function saveCertificate() {
    try {
        const certificate = new Certificate(4, "Rainer Zufall", "CERTIFICATE");
        await PostDocumentRequest.postDocument(certificate);
        successes.push("saveCertificate");
    } catch (e) {
        fails.push("saveCertificate:" + e);
    }
}

async function postDocumentWithConflictingId() {
    try {
        const document = new Document(1);
        await PostDocumentRequest.postDocument(document);
        fails.push("postDocumentWithConflictingId:Request should have led to error but did not.");
    } catch (e) {
        if (e instanceof ApiException) {
            successes.push("postDocumentWithConflictingId");
        } else {
            fails.push("postDocumentWithConflictingId:" + e);
        }
    }
}

await getLetterById();
await getCertificateById();
await getDocumentByUnknownId();
await saveLetter();
await saveCertificate();
await postDocumentWithConflictingId();

const numSuccesses = successes.length;
const numTests = numSuccesses + fails.length;
const successRate = numSuccesses / numTests;

const result = {
    "successful_tests": successes,
    "failed_tests": fails,
    "num_successes": numSuccesses,
    "num_tests": numTests,
    "success_rate": successRate,
}

console.log(TEST_OUTPUT_MARKER);
console.log(JSON.stringify(result));
console.log(TEST_OUTPUT_MARKER);

File: eval/test-cases/inheritance/py/src/tester.py
from generated.api.GetDocumentByIdRequest import GetDocumentByIdRequest
from generated.api.PostDocumentRequest import PostDocumentRequest
from generated.model.Document import Document
from generated.model.Letter import Letter 
from generated.model.Certificate import Certificate 
from generated.exception.ApiException import ApiException

TEST_OUTPUT_MARKER = "#+#"

successes = []
fails = []


def get_letter_by_id():
    try:
        letter = GetDocumentByIdRequest().get_document_by_id(1)
        print(letter)
        if (letter.id == 1):
            successes.append("getLetterById")
        else:
            fails.append("getLetterById:Received letter did not match expected values")
    except Exception as e:
        fails.append(f"getLetterById:{e}")


def get_certificate_by_id():
    try:
        certificate = GetDocumentByIdRequest().get_document_by_id(2)
        if (certificate.id == 2):
            successes.append("getCertificateById")
        else:
            fails.append("getCertificateById:Received certificate did not match expected values")
    except Exception as e:
        fails.append(f"getCertificateById:{e}")


def get_document_by_unknown_id():
    try:
        GetDocumentByIdRequest().get_document_by_id(3)
        fails.append("getDocumentByUnknownId:Request should have led to error but did not.")
    except ApiException:
        successes.append("getDocumentByUnknownId")
    except Exception as e:
        fails.append(f"getDocumentByUnknownId:{e}")


def save_letter():
    try:
        letter = Letter(3, "Mars")
        PostDocumentRequest().post_document(letter)
        successes.append("saveLetter")
    except Exception as e:
        fails.append(f"saveLetter:{e}")


def save_certificate():
    try:
        certificate = Certificate(4, "Rainer Zufall")
        PostDocumentRequest().post_document(certificate)
        successes.append("saveCertificate")
    except Exception as e:
        fails.append(f"saveCertificate:{e}")


def post_document_with_conflicting_id():
    try:
        document = Document(1)
        PostDocumentRequest().post_document(document)
        fails.append("postDocumentWithConflictingId:Request should have led to error but did not.")
    except ApiException:
        successes.append("postDocumentWithConflictingId")
    except Exception as e:
        fails.append(f"postDocumentWithConflictingId:{e}")


# Run the tests
get_letter_by_id()
get_certificate_by_id()
get_document_by_unknown_id()
save_letter()
save_certificate()
post_document_with_conflicting_id()

num_successes = len(successes)
num_tests = num_successes + len(fails)
success_rate = num_successes / num_tests

result = {
    "successful_tests": successes,
    "failed_tests": fails,
    "num_successes": num_successes,
    "num_tests": num_tests,
    "success_rate": success_rate,
}

print(TEST_OUTPUT_MARKER)
print(result)
print(TEST_OUTPUT_MARKER)
